services/worker/src/worker/job_runners/dataset/hub_cache.py

```diff
@@ -72,12 +72,16 @@ def compute_hub_cache_response(dataset: str) -> tuple[DatasetHubCacheResponse, f
             or "dataset" not in content["size"]
             or "num_rows" not in content["size"]["dataset"]
             or not isinstance(content["size"]["dataset"]["num_rows"], int)
+            or not (
+                isinstance(content["size"]["dataset"]["estimated_num_rows"], int)
+                or content["size"]["dataset"]["estimated_num_rows"] is None
+            )
         ):
             raise PreviousStepFormatError(
                 "Previous step 'dataset-size' did not return the expected content: 'partial' or 'size.dataset.num_rows'."
             )
         partial = content["partial"]
-        num_rows = content["size"]["dataset"]["num_rows"]
+        num_rows = content["size"]["dataset"]["estimated_num_rows"] or content["size"]["dataset"]["num_rows"]
```

Comment:
What changes should I do ,

I based this logic on this comment: https://github.com/huggingface/moon-landing/pull/9352#issuecomment-2168158940

Reply:
Maybe I can use `num_rows: int` and `num_rows_origin: Literal["first_5GB", "full_dataset", "estimation"]` ?

Reply:
Yes, I also agreed initially, but I think it would break more than it would help.

For the transition on moonlanding to be easier, we could do one of these:
1. send `num_rows` and `estimated_num_rows`, and handle the logic in moon-landing
2. send `num_rows`, and a new field `num_rows_source` that could take one of these values: `full-exact`, `partial-exact`, `full-estimated`...

I feel like the first one is simpler. cc @julien-c

Reply:
ahah, we wrote at the same time

Reply:
I'd do 2. because `num_rows` is also used for the size tags AFAIK

Reply:
2. sounds good to me

Reply:
as a reference, this is how we use this response (https://github.com/huggingface/moon-landing/blob/0cf8a12a9a85e514b053ef317050077afaad530f/server/repo_types/Datasets.ts#L694C1-L695C44):

```
  viewer:     !hub_cache.viewer ? "preview" : hub_cache.partial ? "viewer-partial" : "viewer",
  numRows:    hub_cache.num_rows,
```

What should we show in the right column when we have a partial conversion + an estimated number of rows for the complete dataset?

If we will always have an estimated number of rows, I have the feeling that we should never show the number of rows of the partial conversion, because it's confusing. And add a warning ("estimated") when we show the estimated number of rows instead of the exact number.

Reply:
Yes showing only the estimation with mention of "estimated" sounds good to me

Reply:
Note that currently, for the side column, we get the info from `dataset-size` (https://github.com/huggingface/moon-landing/blob/0cf8a12a9a85e514b053ef317050077afaad530f/server/lib/DatasetViewer.ts#L84). So, if we don't change this, we don't rely on the contents of dataset-hub-cache. Here, numRows will be used only to:
- compute the size bucket (`1M<n<10M`)
- show the number of rows in the datasets list (`. 15.2M .`)

So... the current code is OK!